### ISSUE TITLE
Bug: InputText's are unusable

### ISSUE THREAD

**Issue by tobix88** [SpaiR/imgui-java#287]
### Version

1.87.3

### What happened?

When pressing SPACE in an InputText that is rendered in a hierarchy of Windows (example below), Space always acts as enter and thus returns true EVEN THOUGH the Window has a ImGuiWindowFlags.NoNavInputs enabled (I suppose it is due to the navigation) and therefore the window always closes which is very frustrating since InputText's can be considered almost a necessity.

### Reproduction

First, I created a general render method (Attention: ImGui is in Minecraft Context!) for all imgui contents like this:
`ImGuiImpl.java`:
```java
    public static void draw(final RenderInterface runnable) {
        imGuiImplGl3.newFrame();
        imGuiImplGlfw.newFrame(); // Handle keyboard and mouse interactions
        ImGui.newFrame();

        runnable.render(ImGui.getIO());

        ImGui.render();
        imGuiImplGl3.renderDrawData(ImGui.getDrawData());

        if (ImGui.getIO().hasConfigFlags(ImGuiConfigFlags.ViewportsEnable)) {
            final long pointer = GLFW.glfwGetCurrentContext();
            ImGui.updatePlatformWindows();
            ImGui.renderPlatformWindowsDefault();

            GLFW.glfwMakeContextCurrent(pointer);
        }
    }
```

`RenderInterface.java`:
```java
import imgui.ImGuiIO;

public interface RenderInterface {

    void render(final ImGuiIO io);

}
```

`ImGuiRenderer.java` (Where everything gets rendered):
```java
public class ImGuiRenderer {
    public static boolean showThisWindow = false;
    public static boolean showThatWindow = false;
    public static boolean showFooWindow = false;
    public static boolean showBarWindow = false;
    ...

    // ▼ Executed with Minecraft Fabric Mixin ▼
    public static void render() {
        if (showThatWindow) ThatWindow.render();
        if (showThisWindow) ThisWindow.render();
        if (showFooWindow) FooWindow.render();
        if (showBarWindow) BarWindow.render();
    }

    ...
}
```

Now, if a TextInput were to be directly rendered in this class, that wouldn't be a problem at all and it works completely fine but as soon as I render it in a second window or upward, it immediately closes the window after pressing space.

Here's an example window:
```java
public class ThatWindow() {

    private static boolean showOtherWindow = false;

    public static void render() {
        if (ImGui.begin("That Window", ImGuiWindowFlags.NoNavInputs)) {
            ImGui.text("Some Text");

            if (ImGui.button("Show Other Window")) showOtherWindow = true;

            if (showOtherWindow) renderOtherWindow();

            ImGui.end();
        }
    }

    ImString imstring = new ImString("", 128);

    public static void renderOtherWindow() {
        if (ImGui.begin("Other Window", ImGuiWindowFlags.NoNavInputs)) {
            ImGui.inputText("##input", imstring);

            ImGui.end();
        }
    }
}
```

Maybe I am doing something wrong, but no matter what I try, it won't resolve. Docking is enabled if that helps (and is an absolute necessity).

**Comment by tobix88** [comment]
If anyone's reading this. IT'S NOT IMGUI'S FAULT! ImGui's doing great, it's Minecraft's fault. Once I fixed this, I am going to close this issue to really, really, really make sure it's Minecraft's fault and then I am going to explain why it's doing this. I unfortunately don't have much time at the moment but I am going to explain this later if anyone's interested and/or has the same issue (which is very unlikely lol).

**Comment by tobix88** [comment]
So, what basically happend is that the button in the background from the Minecraft GUI that opens the ImGui GUI was still focused and whenever you pressed Enter or Space, it got pressed and the ImGui Window opened again thus going back to the first window. I noticed this when I tried if the same thing happens in an actual Minecraft Instance instead of the development environment (where I had sounds turned off) and I heard the typical Minecraft button "click" sound. By then, I knew something was off and I checked the code. Turns out just removing all the children from the Minecraft Screen was not enough and I also had to defocus and focused element to prevent it from being clicked. I've done this by just re-drawing everything so it loses focus and then removing the children. `screen.blur();` was *not* enough apparently.

TL;DR:
Minecraft button was still focused, whenever space or enter was pressed, it activated. By re-drawing everything, it loses focus and then works fine.
